models/Product.js

```diff
@@ -68,7 +68,7 @@ const ProductSchema = new mongoose.Schema(
   },
   { timestamps: true, toJSON: { virtuals: true }, toObject: { virtuals: true } } //timestamps provides fields of createdAt , updatedAt... and exact time, //{ toJSON: { virtuals: true }, toObject: { virtuals: true } } is used in Mongoose to enable virtual fields to be included when converting a document to JSON or a plain JavaScript object
 );
-
+// AKOSREVIEW - I'm curious if the course explained why/when to use a virtual field vs the approach used for the user ref
```

Comment:
Did the instructor give a reason for writing the `reviews` as a virtual field instead of using the `ref` approach like you have for the user field of the product schema?

Reply:
@akosasante he doesn't explain the reason in the videos, so I asked him in comments, if he answers, i'll tell you🙂

Reply:
It's possible he's having y'all do this just to get practice with both approaches. I just usually would use `ref` for anything where the other object I want to embed is a mongoose schema. And then use `virtuals` for fields that need to be calculated/generated by some code (like the subtotal/total of an order for example).